hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/has/isauthorizedraw/IsAuthorizedRawCall.java

```diff
@@ -0,0 +1,266 @@
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.function.Function;
+import org.apache.tuweni.bytes.Bytes;
+import org.hyperledger.besu.evm.frame.MessageFrame;
+import org.hyperledger.besu.evm.gascalculator.GasCalculator;
+import org.hyperledger.besu.evm.precompile.ECRECPrecompiledContract;
+
+/** HIP-632 method: `isAuthorizedRaw` */
+public class IsAuthorizedRawCall extends AbstractCall {
+
+    private final Address address;
+    private final byte[] messageHash;
+    private final byte[] signature;
+
+    private static final GasCalculator noCalculationGasCalculator = new CustomGasCalculator();
+
+    // Besu's ECRECOVER precompile class will be used _directly_ (no need to execute bytecodes):
+    private static final ECRECPrecompiledContract ecPrecompile =
+            new ECRECPrecompiledContract(noCalculationGasCalculator);
+
+    // FUTURE: Gas for system contract method calls needs to be a) determined by measurement of
+    // resources consumed, and b) incorporated into the fee schedule
+    private static final long HARDCODED_GAS_REQUIREMENT_GAS = 1_500_000L;
+
+    public enum SignatureType {
+        Invalid,
+        EC,
+        ED
+    }
+
+    // From Ethereum yellow paper (for reference only):
+    private static BigInteger secp256k1n =
+            new BigInteger("115792089237316195423570985008687907852837564279074904382605163141518161494337");
+
+    public IsAuthorizedRawCall(
+            @NonNull final HasCallAttempt attempt,
+            final Address address,
+            @NonNull final byte[] messageHash,
+            @NonNull final byte[] signature) {
+        super(attempt.systemContractGasCalculator(), attempt.enhancement(), true);
+        this.address = requireNonNull(address);
+        this.messageHash = requireNonNull(messageHash);
+        this.signature = requireNonNull(signature);
+    }
+
+    @NonNull
+    @Override
+    public PricedResult execute(@NonNull final MessageFrame frame) {
+        requireNonNull(frame, "frame");
+
+        final Function<ResponseCodeEnum, PricedResult> bail = rce -> reversionWith(rce, HARDCODED_GAS_REQUIREMENT_GAS);
+
+        // Fail fast if user didn't supply monstrous amount of gas
+        final var availableGas = frame.getRemainingGas();
+        if (availableGas < HARDCODED_GAS_REQUIREMENT_GAS) return bail.apply(INSUFFICIENT_GAS);
+
+        // Figure out what kind of signature we're dealing with thus what kind of account+key we need
+        final var signatureType =
+                switch (signature.length) {
+                    case 65 -> SignatureType.EC;
+                    case 64 -> SignatureType.ED;
+                    default -> SignatureType.Invalid;
+                };
+        if (signatureType == SignatureType.Invalid)
+            return bail.apply(FAIL_INVALID /* should be: "invalid argument to precompile" */);
```

Comment:
⚠️ `FAIL_INVALID` should be reserved for internal system errors, not user errors.

Reply:
Will use `INVALID_TRANSACTION_BODY` as you suggested to me elsewhere.